src/App.jsx

```diff
@@ -1,28 +1,91 @@
-import React, { useState } from 'react';
+import React, { useState, useRef, useEffect } from 'react';
 import { DragDropContext, Droppable, Draggable } from 'react-beautiful-dnd';
+import EmojiPicker from 'emoji-picker-react';
 import './App.css';
 
 const initialColumns = {
   todo: {
     id: 'todo',
     title: 'To Do',
-    tasks: []
-  },
-  inProgress: {
-    id: 'inProgress',
-    title: 'In Progress',
-    tasks: []
-  },
-  done: {
-    id: 'done',
-    title: 'Done',
-    tasks: []
+    tasks: [],
+    order: 0,
+    emoji: '📝',
+    color: '#1a73e8'
   }
 };
 
 function App() {
   const [columns, setColumns] = useState(initialColumns);
   const [newTask, setNewTask] = useState('');
+  const [newColumnTitle, setNewColumnTitle] = useState('');
+  const [boardTitle, setBoardTitle] = useState('Kanban Board');
+  const [isEditingTitle, setIsEditingTitle] = useState(false);
+  const titleInputRef = useRef(null);
+  const [showEmojiPicker, setShowEmojiPicker] = useState(null);
+  const [showColorPicker, setShowColorPicker] = useState(null);
+  const emojiPickerRef = useRef(null);
+
+  useEffect(() => {
+    if (isEditingTitle && titleInputRef.current) {
+      titleInputRef.current.focus();
+      titleInputRef.current.select();
+    }
+  }, [isEditingTitle]);
+
+  useEffect(() => {
+    function handleClickOutside(event) {
+      if (emojiPickerRef.current && !emojiPickerRef.current.contains(event.target)) {
+        setShowEmojiPicker(null);
+        setShowColorPicker(null);
+      }
+    }
+
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => document.removeEventListener('mousedown', handleClickOutside);
+  }, []);
+
+  const handleTitleClick = () => {
+    setIsEditingTitle(true);
+  };
+
+  const handleTitleChange = (e) => {
+    setBoardTitle(e.target.value);
+  };
+
+  const handleTitleBlur = () => {
+    setIsEditingTitle(false);
+    if (!boardTitle.trim()) {
+      setBoardTitle('Kanban Board');
+    }
+  };
+
+  const handleTitleKeyDown = (e) => {
+    if (e.key === 'Enter') {
+      handleTitleBlur();
+    }
+  };
+
+  const handleAddColumn = (e) => {
+    e.preventDefault();
+    if (!newColumnTitle.trim()) return;
+
+    const columnId = newColumnTitle.toLowerCase().replace(/\s+/g, '-');
+    const maxOrder = Math.max(...Object.values(columns).map(col => col.order), -1);
+    
+    setColumns({
+      ...columns,
+      [columnId]: {
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```suggestion
const generateUniqueId = (existingTasks) => {
  const newId = `task-${Date.now()}`;
  return existingTasks.some(task => task.id === newId)
    ? `${newId}-${Math.random().toString(36).substr(2, 9)}`
    : newId;
};

const task = {
  id: generateUniqueId(columns[firstColumnId]?.tasks || []),
  content: newTask
};
```
The handleAddTask function should validate the task ID to ensure uniqueness. Currently, there's no mechanism to prevent duplicate task IDs which could cause issues with React keys and drag-and-drop functionality.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;